### DISCUSSION TITLE
Stride website in blazor

### DISCUSSION THREAD

**Discussion by ykafia** [stride3d/stride#1220] [Contributors]
I'm having issues with the whole jekyll thingy with Stride, i was thinking we could rewrite the website in blazor since everyone must has a dotnet toolchain somewhere and it would be a `dotnet watch` away from debugging.
What do you guys think ?

**Comment by manio143** [comment]
I'd be against that in general. Jekyll is widely used in the blogosphere, rewriting the website takes time, I don't know if blazor has a static site generator - with Jekyll you compile the website into pure html+css+js package that has nothing to do with Jekyll and can be hosted anywhere. I understand that on Windows ruby and jekyll can be problematic and I know there's alternatives such as Hugo which are distributed in a single executable. But again, changing to any other site generator will involve a lot of changes.

**Reply by ykafia** [reply]
What jekyll offers is also available in .NET as a nuget (Fsharp.Formating for example), the static site generation is available in blazor in the end. 

But I agree, it would be a complex task to rewrite the websites in blazor.

**Comment by VaclavElias** [comment]
I also thought about using ASP.NET Core and Blazor which I use often but I don't know anything about blogosphere and consequences if the Stride website was changed for this stack. 

One advantage is that our community is C# experienced so more people could be participating.

Also, I think there was a discussion regarding filtering NuGet packages for Stride assets and I think it would be easy to implement in Blazor, so Stride website could have some simple version and showing assets from NuGet, and if this goes well some simple API could be done and read it also from Stride Studio. Actually, this could be done regardless of current implementation.

Other option could be simplifying the layout and using also md files for certain content (pages, tutorials) for writing the content, so even GitHub could be used for writing/correcting these pages and then the site could be generated by current implementation or ASP.NET Core/Blazor. 

Microsoft .NET Docs are done this way. Example: https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/web-api

In any case as you said, it is lots of work....

If you ever decided doing it, I would suggest to do it gradually, page by page, so some pages can use currently implementation while the other pages would use the new one..

**Reply by ykafia** [reply]
The website is static so we would just need blazor for the frontend 🙂 
I'm working on a prototype somehow, will see how doable everything is

**Comment by Aggror** [comment]
Using Blazor for a Stride nuget package browser (free marketplace) would be nice. That way we can integrate it in the launcher or editor.
